datafusion/core/src/datasource/listing/table.rs

```diff
@@ -254,6 +249,26 @@ impl ListingOptions {
             file_sort_order: None,
         }
     }
+
+    /// Build ListingOptions from an existing set of ListingOptions
```

Comment:
I think typically in rust one uses `options.clone()` for this operation

I will leave a comment at the usage site

Reply:
yep, that makes a lot more sense!